WordPress/Classes/ViewRelated/Reader/Detail/Views/ReaderDetailFeaturedImageView.swift

```diff
@@ -259,14 +274,39 @@ class ReaderDetailFeaturedImageView: UIView, NibLoadable {
             return
         }
 
+        loadingView.isHidden = false
+
         isLoading = true
         isLoaded = true
 
-        imageLoader.loadImage(with: imageURL, from: post, placeholder: nil, success: { [weak self] in
-            self?.didFinishLoading()
+        let completionHandler: (CGSize) -> Void = { [weak self] size in
+            guard let self = self else {
+                return
+            }
+
+            self.imageSize = size
+            self.didFinishLoading()
+            self.isLoading = false
```

Comment:
There are two remaining issues here:

* Saved posts (which are meant to work while offline or in a bad connection)
* Users in a really bad connection

I guess both cases can be solved by adding a timeout. It probably won't affect the current functionality. Wdyt?

Reply:
I agree, how long of a timeout do you think we should use?

Reply:
3 or 4 seconds? I thought about 5 but it seems a lot, and in most cases (even in bad connections) the image size will load in less than a second, right?

Reply:
Done in: https://github.com/wordpress-mobile/WordPress-iOS/pull/14884/commits/0aa3f5080b86c8eedca091b3c6414656ae856ce7